### ISSUE TITLE
`PrepareQuery` callback never firing

### ISSUE THREAD

**Issue by tanner0101** [vapor/mysql#126]
Tracking down an issue where `model.save(...)` is never completing (for a complex route).

Looks like `callback` parameter to `PrepareQuery` is never firing.

![screen shot 2018-02-19 at 7 14 48 pm](https://user-images.githubusercontent.com/1342803/36402005-5f8467c0-15a9-11e8-805f-a5771e0409f6.png)
![screen shot 2018-02-19 at 7 14 53 pm](https://user-images.githubusercontent.com/1342803/36402006-61cf31a4-15a9-11e8-8a04-9e8e16d4f1eb.png)

Expected output would be:
```
START preparing
done preparing
```

[hangtest.zip](https://github.com/vapor/mysql/files/1738565/hangtest.zip)

Zip file attached can be used to re-create. Note you will need to add this hack:

```swift
extension Router {
    /// Creates a `Route` at the provided path using the `PUT` method.
    ///
    /// [Learn More →](https://docs.vapor.codes/3.0/getting-started/routing/)
    @discardableResult
    func get<C, T>(
        _ content: C.Type,
        at path: PathComponent...,
        use closure: @escaping RequestDecodableResponder<C, T>.Closure
    ) -> Route<Responder> where C: RequestDecodable, T: ResponseEncodable {
        let responder = RequestDecodableResponder(closure: closure)
        let route = Route<Responder>(
            path: [.constants([.bytes(HTTPMethod.get.bytes)])] + path,
            output: responder
        )
        self.register(route: route)
        return route
    }
}
```

You will also need `test_user:test_user@testdb` configured as a MySQL user w/ full rights.

**Comment by SandorDobi** [comment]
this is fixed here @ #127

**Comment by Joannis** [comment]
Merged #127
